Terminal.Gui/ConsoleDrivers/AnsiResponseParser/AnsiResponseParser.cs

```diff
@@ -1,13 +1,41 @@
 ﻿#nullable enable
 
+using System.Runtime.ConstrainedExecution;
+using Microsoft.Extensions.Logging;
+
 namespace Terminal.Gui;
 
 internal abstract class AnsiResponseParserBase : IAnsiResponseParser
 {
+    private const char Escape = '\x1B';
+    private readonly AnsiMouseParser _mouseParser = new ();
+    protected readonly AnsiKeyboardParser _keyboardParser = new ();
     protected object _lockExpectedResponses = new ();
 
     protected object _lockState = new ();
 
+    /// <summary>
+    ///     Event raised when mouse events are detected - requires setting <see cref="HandleMouse"/> to true
+    /// </summary>
+    public event EventHandler<MouseEventArgs>? Mouse;
+
+    /// <summary>
+    ///     Event raised when keyboard event is detected (e.g. cursors) - requires setting <see cref="HandleKeyboard"/>
+    /// </summary>
+    public event Action<object, Key>? Keyboard;
```

Comment:
Why Action? I thought we agreed on this model for events in the library?

https://gui-cs.github.io/Terminal.GuiV2Docs/docs/events.html

Specifically:

> Common Patterns
> The primary pattern for events is the event/EventHandler idiom. We use the Action<T> idiom > sparingly. We support INotifyPropertyChanged in cases where data binding is relevant.

Happy to be educated on why Action is better here.

Reply:
Just an oversight, have fixed 🚀